util/config/fixpath.py

```diff
@@ -1,80 +1,94 @@
-    4. $CHPL_HOME is set, but path doesn't include an old one.
-       ($CHPL_HOME was hand-set, now setchplenv is run)
+1. No Chapel environment settings (new shell)
+2. Same $CHPL_HOME as last time (re-running setchplenv in same dir)
+3. Different $CHPL_HOME (cd ../other-chapel-dir).
+4. $CHPL_HOME is set, but path doesn't include an old one.
+   ($CHPL_HOME was hand-set, now setchplenv is run)
 
-    For case 1, just return the existing environment variable.
+For case 1, just return the existing environment variable.
 
-    For case 2, return the environment variable without the components
-    that begin with $CHPL_HOME.
+For case 2, return the environment variable without the components
+that begin with $CHPL_HOME.
 
-    For case 3, setchplenv invokes this script before setting the new
-    $CHPL_HOME.  We still have the old $CHPL_HOME set, so we can remove
-    the old $PATH and $MANPATH entries.  The upshot is we do the same thing
-    as in case 2.
+For case 3, setchplenv invokes this script before setting the new
+$CHPL_HOME.  We still have the old $CHPL_HOME set, so we can remove
+the old $PATH and $MANPATH entries.  The upshot is we do the same thing
+as in case 2.
+
+Case 4 should also be the same as case 2, but we won't remove any
+components since there should be no components starting with $CHPL_HOME.
+Mentioned only to avoid reintroducing #10196 when this function is modified.
+"""
+
+import optparse
+import os
+import re
+import sys
+
+def escape_path(p, delim):
+    """Wrap fish paths in quotes to prevent splitting on spaces in paths"""
+    if delim == ' ':
+        return '"{}"'.format(p)
+    return p
 
-    Case 4 should also be the same as case 2, but we won't remove any
-    components since there should be no components starting with $CHPL_HOME.
-    Mentioned only to avoid reintroducing #10196 when this function is modified.
 
-    :env: path environment variable ('PATH' or 'MANPATH')
+def remove_chpl_from_path(path_val, delim):
+    """
+    :path_val: path environment variable value ('$PATH' or '$MANPATH')
     :delim: path delimiter (':' or ' ')
     :returns: new path with $CHPL_HOME components removed
     """
 
-    # Get environment variables, $(MAN)PATH will be always be ':' delimited
-    path = getenv(env)
-    if path is None:
-        return ''
+    chpl_home = os.getenv('CHPL_HOME')
 
-    chpl_home = getenv('CHPL_HOME')
+    if not chpl_home or chpl_home not in path_val:
+        return path_val
 
-    # Find ':'s that are not escaped
+    # Find delims that are not escaped
+    # Note: Fish shell still uses ':' delimiter for printing with \\"$PATH\\"
     pattern = r'(?<!\\)\:'
 
-    # Split path into list separated by non-escaped ':'s, and sieve chpl_home
-    newpath = [escape_path(p)
-                 for p in re.split(pattern, path)
-                     if chpl_home is None or chpl_home not in p]
+    # Split path by non-escape delimiters and sieve chpl_home
```

Comment:
Given the above comment, seems like this should go back to `# Split path by by non-escaped ':'s, and sieve chpl_home`